addons/hr_holidays/models/hr_employee.py

```diff
@@ -385,9 +385,9 @@ def _get_mandatory_days(self, start_date, end_date):
     @api.model
     def _get_contextual_employee(self):
         ctx = self.env.context
-        if 'employee_id' in ctx:
+        if 'employee_id' in ctx and self.env.context.get('employee_id'):
             return self.browse(ctx.get('employee_id'))
-        if 'default_employee_id' in ctx:
+        if 'default_employee_id' in ctx and self.env.context.get('default_employee_id'):
```

Comment:
could be written

```suggestion
        if self.env.context.get('employee_id'):
            return self.browse(ctx.get('employee_id'))
        if self.env.context.get('default_employee_id'):
```

Reply:
not sure I understand, the meaning of your commit is I think:

- if there is no `employee_id` in context, or if it has value False or None
- and if there is no `default_employee_id` in context, of if it has value False or None

=> then fallback to `self.env.user.employee_id`

when before your commit, it was only if there was no `employee_id` or `default_employee_id` that we would fallback to `self.env.user.employee_id` even if the value was falsy.

Here I'm just suggesting an equivalent code:

```py
if 'employee_id' in self.env.context and self.env.context.get('employee_id'):
```
and:

```py
if self.env.context.get('employee_id'):
```
should always be equivalent code (because if `employee_id` is not in the dictionary, the `get` function returns a `None` value which is falsy.